### ISSUE TITLE
Setup Error

### ISSUE THREAD

**Issue by yoalley** [OrchardCMS/OrchardCore#6585]
System.IO.DirectoryNotFoundException
  HResult=0x80070003
  Message=Could not find a part of the path 'F:\Git\net core\cms\OrchardCore\src\OrchardCore.Cms.Web\App_Data\Sites\Default\ContentDefinition.json'.
  Source=System.Private.CoreLib
  StackTrace:
   在 System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   在 System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   在 System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   在 System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   在 System.IO.StreamWriter..ctor(String path, Boolean append)
   在 System.IO.File.CreateText(String path)
   在 OrchardCore.ContentManagement.FileContentDefinitionStore.SaveContentDefinitionAsync(ContentDefinitionRecord contentDefinitionRecord) 在 F:\Git\net core\cms\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement\FileContentDefinitionStore.cs 中: 第 68 行
   在 OrchardCore.ContentManagement.ContentDefinitionManager.UpdateContentDefinitionRecord() 在 F:\Git\net core\cms\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement\ContentDefinitionManager.cs 中: 第 348 行
   在 OrchardCore.ContentManagement.ContentDefinitionManager.StorePartDefinition(ContentPartDefinition contentPartDefinition) 在 F:\Git\net core\cms\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement\ContentDefinitionManager.cs 中: 第 132 行
   在 OrchardCore.ContentManagement.Metadata.ContentDefinitionManagerExtensions.AlterPartDefinition(IContentDefinitionManager manager, String name, Action`1 alteration) 在 F:\Git\net core\cms\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement.Abstractions\IContentDefinitionManager.cs 中: 第 60 行
   在 OrchardCore.Liquid.Migrations.Create() 在 F:\Git\net core\cms\OrchardCore\src\OrchardCore.Modules\OrchardCore.Liquid\Migrations.cs 中: 第 18 行

**Comment by hishamco** [comment]
Could you please share the steps to reproduce? Which version you are using?

**Comment by sebastienros** [comment]
Closing until more repro steps are provided

**Comment by rserj** [comment]
I had the same issue, I run orchardcore in a docker container (latest dev "orchardproject/orchardcore-cms-linux:dev" via AWS Fargate) I'm installing one by using MySql and Agency recipe. After I hit the install button I got the same error

`An unhandled exception occurred while processing the request. DirectoryNotFoundException: Could not find a part of the path '/app/App_Data/Sites/Default/ContentDefinition.json'.`

But when I install "Software as a service recipe" it goes fine, then I login into the admin dashboard and run Agency recipe it goes fine too. So only Agency recipe on first installation fails

**Comment by dkoshy** [comment]
I also received same issue when configuring for Agency recipe .

Thank You.

**Comment by mkizito76** [comment]
I am not sure if this issue is fixed or not. I ran into the issue multiple times when using the agency recipe (and also a custom recipe).
Steps:
1. I followed steps to [create a decoupled CMS site](https://docs.orchardcore.net/en/dev/docs/guides/decoupled-cms/) and even used the steps in [creating an orchardcore site](https://docs.orchardcore.net/en/dev/docs/guides/create-cms-application/)
2. Note that I am using SQL Server for all the steps 
3. That worked as expected and I created an empty site using the blank recipe. I then deleted the database and App_Data folder 
3. Created a custom them with a recipe added the reference to the main project and got the error below when trying to use the custom recipe for the site set up
4. I changed the nuget package to "OrchardCore.Application.Cms.Core.Targets" and could create a new website using the Blog recipe
5. I tried again with the custom recipe and it not work. I tried using the agency recipe and got the same error
6. I copied the blog and agency recipe to my custom theme and tried again (after deleted the database and App_Data folders). The Blog recipe worked but the agency and custom recipes failed
7. I then downloaded the full source code (OrchardCore-1.0.0-rc2) and tried to run the agency recipe and got the same error
Below is the fill error I got when I used the source code
`System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\<username>\source\repos\Orchard.CMS.Web\App_Data\Sites\Default\ContentDefinition.json'.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.StreamWriter..ctor(String path, Boolean append)
   at System.IO.File.CreateText(String path)  at OrchardCore.ContentManagement.FileContentDefinitionStore.SaveContentDefinitionAsync(ContentDefinitionRecord contentDefinitionRecord) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ContentManagement\FileContentDefinitionStore.cs:line 68
   at OrchardCore.ContentManagement.ContentDefinitionManager.UpdateContentDefinitionRecord() in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ContentManagement\ContentDefinitionManager.cs:line 348
   at OrchardCore.ContentManagement.ContentDefinitionManager.StoreTypeDefinition(ContentTypeDefinition contentTypeDefinition) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ContentManagement\ContentDefinitionManager.cs:line 127
   at OrchardCore.ContentManagement.Metadata.ContentDefinitionManagerExtensions.AlterTypeDefinition(IContentDefinitionManager manager, String name, Action1 alteration) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.ContentManagement.Abstractions\IContentDefinitionManager.cs:line 54
   at OrchardCore.ContentTypes.RecipeSteps.ContentDefinitionStep.UpdateContentType(ContentTypeDefinition type, ContentTypeDefinitionRecord record) in C:\projects\orchardcore\src\OrchardCore.Modules\OrchardCore.ContentTypes\RecipeSteps\ContentDefinitionStep.cs:line 66
   at OrchardCore.ContentTypes.RecipeSteps.ContentDefinitionStep.ExecuteAsync(RecipeExecutionContext context) in C:\projects\orchardcore\src\OrchardCore.Modules\OrchardCore.ContentTypes\RecipeSteps\ContentDefinitionStep.cs:line 37
   at OrchardCore.Recipes.Services.RecipeExecutor.<>c__DisplayClass9_0.<<ExecuteStepAsync>b__0>d.MoveNext() in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 173
--- End of stack trace from previous location where exception was thrown ---
   at OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func`2 execute) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Abstractions\Shell\Scope\ShellScope.cs:line 206
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteStepAsync(RecipeExecutionContext recipeStep) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 154
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, Object environment, CancellationToken cancellationToken) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 100
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, Object environment, CancellationToken cancellationToken) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 117
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, Object environment, CancellationToken cancellationToken)
   at OrchardCore.Setup.Services.SetupService.SetupInternalAsync(SetupContext context) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Setup.Core\SetupService.cs:line 190
   at OrchardCore.Setup.Services.SetupService.SetupAsync(SetupContext context) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Setup.Core\SetupService.cs:line 85
   at OrchardCore.Setup.Controllers.SetupController.IndexPOST(SetupViewModel model) in C:\projects\orchardcore\src\OrchardCore.Modules\OrchardCore.Setup\Controllers\SetupController.cs:line 170
   at lambda_method(Closure , Object )
   at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at OrchardCore.Modules.ModularTenantRouterMiddleware.Invoke(HttpContext httpContext) in C:\projects\orchardcore\src\OrchardCore\OrchardCore\Modules\ModularTenantRouterMiddleware.cs:line 63
   at OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func`2 execute) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Abstractions\Shell\Scope\ShellScope.cs:line 206
   at OrchardCore.Modules.ModularTenantContainerMiddleware.Invoke(HttpContext httpContext) in C:\projects\orchardcore\src\OrchardCore\OrchardCore\Modules\ModularTenantContainerMiddleware.cs:line 59
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)`
